src/phase0/state_transition/mod.rs

```diff
@@ -1009,3 +1004,172 @@ pub fn get_attesting_indices<
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+>(
+    state: &mut BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    index: ValidatorIndex,
+    context: &Context,
+) {
+    if state.validators[index].exit_epoch != FAR_FUTURE_EPOCH {
+        return;
+    }
+
+    let mut exit_epochs: Vec<Epoch> = state
+        .validators
+        .as_slice()
+        .iter()
+        .filter(|v| v.exit_epoch != FAR_FUTURE_EPOCH)
+        .map(|v| v.exit_epoch)
+        .collect();
+
+    exit_epochs.push(compute_activation_exit_epoch(
+        get_current_epoch(state, context),
+        context,
+    ));
+
+    let mut exit_queue_epoch = *exit_epochs.iter().max().unwrap();
+
+    let exit_queue_churn = state
+        .validators
+        .as_slice()
+        .iter()
+        .filter(|v| v.exit_epoch == exit_queue_epoch)
+        .count();
+
+    if exit_queue_churn >= get_validator_churn_limit(state, context) as usize {
+        exit_queue_epoch += 1
+    }
+
+    state.validators[index].exit_epoch = exit_queue_epoch;
+    state.validators[index].withdrawable_epoch =
+        state.validators[index].exit_epoch + context.min_validator_withdrawability_delay as u64;
+}
+
+pub fn slash_validator<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+>(
+    state: &mut BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    slashed_index: ValidatorIndex,
+    whistleblower_index: Option<ValidatorIndex>,
+    context: &Context,
+) -> Result<(), Error> {
+    let epoch = get_current_epoch(state, context);
+    initiate_validator_exit(state, slashed_index, context);
+    state.validators[slashed_index].slashed = true;
+    state.validators[slashed_index].withdrawable_epoch = u64::max(
+        state.validators[slashed_index].withdrawable_epoch,
+        epoch + context.epochs_per_slashings_vector as u64,
+    );
+    state.slashings[epoch as usize % EPOCHS_PER_SLASHINGS_VECTOR as usize] +=
```

Comment:
imo a bit more readable to move the expression inside the `[..]` to the prior line as a `slashings_index` or something like that

Reply:
fixed